hack/benchmark/internal/client/ngtd/grpc/client.go

```diff
@@ -46,389 +48,459 @@ func New(ctx context.Context, opts ...Option) (Client, error) {
 		opt(c)
 	}
 
-	c.Client = grpc.New(c.opts...)
+	c.c = grpc.New(c.opts...)
 
-	if err := c.Client.Connect(ctx, c.addr); err != nil {
+	if err := c.c.Connect(ctx, c.addr); err != nil {
 		return nil, err
 	}
 
 	return c, nil
 }
 
-func (c *ngtdClient) Exists(
-	ctx context.Context,
-	req *client.ObjectID,
-) (*client.ObjectID, error) {
-	return nil, errors.ErrUnsupportedClientMethod
+func (c *ngtdClient) Exists(ctx context.Context, in *payload.Object_ID, opts ...grpc.CallOption) (oid *payload.Object_ID, err error) {
+	_, err = c.c.Do(ctx, c.addr, func(ctx context.Context,
+		conn *grpc.ClientConn,
+		copts ...grpc.CallOption) (interface{}, error) {
+		id, err := proto.NewNGTDClient(conn).GetObject(ctx, &proto.GetObjectRequest{
+			Id: []byte(in.GetId()),
+		})
+		if err != nil {
+			return nil, err
+		}
+		oid = &payload.Object_ID{
+			Id: string(id.GetId()),
+		}
+		return oid, nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	return oid, nil
 }
 
-func (c *ngtdClient) Search(
-	ctx context.Context,
-	req *client.SearchRequest,
-) (*client.SearchResponse, error) {
-	res, err := c.Client.Do(ctx, c.addr,
-		func(ctx context.Context, conn *grpc.ClientConn, copts ...grpc.CallOption) (interface{}, error) {
-			res, err := proto.NewNGTDClient(conn).Search(ctx, searchRequestToNgtdSearchRequest(req), copts...)
-			if err != nil {
-				return nil, err
-			}
-
-			if len(res.GetError()) != 0 {
-				return nil, errors.New(res.GetError())
-			}
-			return res, nil
-		},
-	)
+func (c *ngtdClient) Search(ctx context.Context, in *payload.Search_Request, opts ...grpc.CallOption) (*payload.Search_Response, error) {
+	res, err := c.c.Do(ctx, c.addr, func(ctx context.Context,
+		conn *grpc.ClientConn,
+		copts ...grpc.CallOption) (interface{}, error) {
+		return proto.NewNGTDClient(conn).Search(ctx, searchRequestToNgtdSearchRequest(in), copts...)
+	})
```

Comment:
should we check error as bellow link?
or should we delete error checking of `Insert` method like `Search` method?



https://github.com/vdaas/vald/pull/803/files#diff-c10c54d8075bfe0dfd619a6ebee74f91cdab16dd83b3a51e4efbe6f416073e9bR194-R196

Reply:
what do you mean??

Reply:
I can't see your line link

Reply:
ping

Reply:
sorry. The bellow is line link.
https://github.com/vdaas/vald/blob/0a4e993519a4c3271fd4eed81bd0ead12312e244/hack/benchmark/internal/client/ngtd/grpc/client.go#L194-L196

Because ʻInsert` was handling errors, I was wondering if we should also do it with` Search`, so I asked.

Reply:
got it.
fixed